extensions/twitter/src/timeline.tsx

```diff
@@ -0,0 +1,108 @@
+import { ActionPanel, Image, ImageMask, List, OpenInBrowserAction, showToast, ToastStyle } from "@raycast/api";
+import { useState, useEffect } from "react";
+import { TweetV1 } from "twitter-api-v2";
+import { twitterClient } from "./twitterapi";
+
+export default function TweetList() {
+  const { data, error, isLoading } = useSearch("");
+  if (error) {
+    showToast(ToastStyle.Failure, "Error", error);
+  }
+  return (
+    <List isLoading={isLoading} searchBarPlaceholder="Filter Tweets by name...">
+      {data?.map((tweet) => (
+        <TweetListItem key={tweet.id_str} tweet={tweet} />
+      ))}
+    </List>
+  );
+}
+
+function TweetListItem(props: { tweet: TweetV1 }) {
+  const t = props.tweet;
+
+  const maxLength = 80;
+  const textRaw = t.full_text ? t.full_text.trim() : "";
+  const text = textRaw.slice(0, maxLength) + (textRaw.length > maxLength ? " ..." : "");
+
+  const imgUrl = t.user.profile_image_url_https;
+  const icon: Image | undefined = imgUrl
+    ? { source: t.user.profile_image_url_https, mask: ImageMask.Circle }
+    : undefined;
+
+  const tweetUrl = `https://twitter.com/${t.user.screen_name}/status/${t.id_str}`;
+  const urls = t.entities.urls;
+  const hasImage = urls && urls.length > 0;
+  let states = [`💬 ${t.reply_count || 0}`, `🔁 ${t.retweet_count}`, `❤️ ${t.favorite_count}`];
+  if (hasImage) {
+    states = ["🖼️", ...states];
+  }
+
+  return (
+    <List.Item
+      id={t.id_str}
+      key={t.id_str}
+      title={text}
+      icon={icon}
+      accessoryTitle={states.join("  ")}
+      actions={
+        <ActionPanel>
+          <OpenInBrowserAction url={tweetUrl} />
+        </ActionPanel>
+      }
+    />
+  );
+}
+
+export function useSearch(query: string | undefined): {
+  data: TweetV1[] | undefined;
+  error?: string;
+  isLoading: boolean;
+} {
+  const [data, setData] = useState<TweetV1[]>();
+  const [error, setError] = useState<string>();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```

Comment:
Thanks for the hint. It seems that react-query can not store data to disk. Is there any mechanism?

Reply:
https://react-query.tanstack.com/plugins/persistQueryClient#_top it can serialize to JSON, and then you determine how to persist to disk. I don't know what the raycast-recommended way to do this would be (I'm interested for my extension too!), @nhojb ?